addons/survey/views/survey_templates.xml

```diff
@@ -421,13 +421,14 @@
     <template id="question_simple_choice" name="Question: simple choice">
```

Comment:
Here is my suggestion after giving it a few thoughts, trying to simplify the code and match the spec. It would become something like this (it covers my other comments as well):

```xml
<template id="question_simple_choice" name="Question: simple choice">
...
    <t t-set="answers_contains_image" t-value="question.suggested_answer_ids.filtered(lambda x : x.value_image)"/>
    <t t-set="use_half_columns" t-value="survey.questions_layout != 'page_per_question' or answers_contains_image"/>
    <t t-set="use_single_column" t-value="use_half_columns and not answers_contains_image"/>
    <div class="row g-2 o_survey_form_choice"
         t-att-data-name="question.id"
         data-question-type="simple_choice_radio">
        ...
        <t t-foreach='question.suggested_answer_ids' t-as='label'>
            ...
            <div t-att-class="'col-sm-12 %s' % ('col-lg-6' if use_half_columns else '')">
                <label t-att-for="str(question.id) + '_' + str(label.id)"
                    t-att-class="'o_survey_choice_btn py-1 px-3 w-100 h-100 rounded %s %s' % (answer_class, 'o_survey_selected' if answer_selected else '')">
                ...
                </label>
            </div>
            <!--Go to line for vertical list of answers -->
            <div t-if="use_single_column" class="w-100 m-0"/>
        </t>
        <t t-if='question.comments_allowed and question.comment_count_as_answer'>
            <div t-att-class="'col-sm-12 %s' % ('col-lg-6' if use_half_columns else '')">
                <label t-att-class="'o_survey_choice_btn py-1 px-3 h-100 w-100 rounded %s' % ('o_survey_selected' if comment_line else '')">
                ...
                </label>
            </div>
            <!--Go to line for vertical list of answers -->
            <div t-if="use_single_column" class="w-100 m-0"/>
            <div t-attf-class="o_survey_comment_container mt-3 py-0 px-1  #{'d-none' if not comment_line else ''}">
                <textarea type="text" class="form-control o_survey_question_text_box bg-transparent text-dark rounded-0 p-0"
                          t-att-disabled="None if comment_line else 'disabled'"><t t-esc="comment_line.value_char_box if comment_line else ''"/></textarea>
            </div>
        </t>
        <div t-if='question.comments_allowed and not question.comment_count_as_answer'
            t-attf-class="mb-2 o_survey_comment_container mt-3 #{'col-lg-6' if use_single_column else ''}">
            <textarea type="text" class="col form-control o_survey_comment o_survey_question_text_box bg-transparent text-dark rounded-0 p-0"
                      t-att-placeholder="question.comments_message or default_comments_message if not survey_form_readonly else ''"><t t-esc="comment_line.value_char_box if comment_line else ''"/></textarea>
        </div>
    </div>
</template>
```

A few comments:
- Row is back around both answers and comment_answer. The wole question would only be one row, which is quite simple.
- The readability is improved with the few t-set variables at the start of the template. The advantage is that one can use them both in the answers and comment_answer styling.
- some div become t to avoid going to the line.
- g-2 on the row will do most of the spacing between elements :)

Of course this would be done as well in the multiple choice question